### PR TITLE
Refactoring Conditional to BooleanDefaultTrue

### PR DESCRIPTION
**Pull request by yunhee-l**
### Summary
Refactoring Conditional to BooleanDefaultTrue, replacing "DefaultTrue" to "NotSet", so that default false values can be supported.  
This is the first step in fixing the boolean config merge bug; I'll implement BooleanDefaultFalse next and apply it to one boolean config + test, and after that we need to migrate all boolean configs to use either of BooleanDefaultTrue or BooleanDefaultFalse instead of just boolean. I'm breaking down these to have reasonably sized PRs. 

### Implementation details
Replaced DefaultTrue with NotSet, and moved interpretation of NotSet value to a new wrapper struct BooleanDefaultTrue. With this, BooleanDefaultFalse can be implemented to use the same enum values but apply different defaulting logic to the not set boolean config.

### Testing

New tests cover the changes: 

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: agent/config/config_unix.go [hunk 1]
```diff
@@ -71,7 +71,7 @@ func DefaultConfig() Config {
 		PauseContainerTag:                   DefaultPauseContainerTag,
 		AWSVPCBlockInstanceMetdata:          false,
```
Comment (sparrc) [inline agent/config/config_unix.go:72]: Why are we not implementing a BooleanDefaultFalse struct and replacing all of these config vars that default to false with that? Are "default false" config vars not affected by this bug?

**Comment by sparrc** [comment]
ECS_POLL_METRICS also defaults to true (see https://github.com/aws/amazon-ecs-agent/blob/master/agent/config/config.go#L551), does this env var also need to be replaced with BooleanDefaultTrue?

**Comment by yunhee-l** [comment]
> ECS_POLL_METRICS also defaults to true (see https://github.com/aws/amazon-ecs-agent/blob/master/agent/config/config.go#L551), does this env var also need to be replaced with BooleanDefaultTrue?

Yes eventually. Goal is to add BooleanDefaultFalse next, and migrate the bool configs to use these.

File: agent/config/config_unix.go (hunk 1 above)
Comment (yunhee-l) [reply agent/config/config_unix.go:72]: im working on that now, each change would be 20+ file changes so I am breaking them down.

**Pull request closed without merging**